server/src/test/java/io/spine/server/bus/BusesShould.java

```diff
@@ -51,4 +73,140 @@ public void not_accept_nulls() {
                                                       .build())
                 .testAllPublicStaticMethods(Buses.class);
     }
+
+    @Test
+    public void acknowledged_valid_message_delivery() {
+        final TestMessageBus bus = busBuilder().build();
+        final BusMessage message = busMessage(testContents());
+        final MemoizingObserver<Ack> observer = memoizingObserver();
+
+        bus.post(message, observer);
+
+        final List<Ack> responses = observer.responses();
+        assertSize(1, responses);
+
+        final Ack response = responses.get(0);
+        assertEquals(STATUS_OK, response.getStatus());
+    }
+
+    @Test
+    public void apply_the_validating_filter() {
+        final TestMessageBus bus = busBuilder().failingValidation()
+                                               .build();
+
+        assertBusPostErrs(bus, ErrorType.FAILING_VALIDATION);
+    }
+
+    @Test
+    public void apply_the_dead_message_filter() {
+        final TestMessageBus bus = busBuilder().withNoDispatchers()
+                                               .build();
+
+        assertBusPostErrs(bus, ErrorType.DEAD_MESSAGE);
+    }
+
+    @Test
+    public void apply_a_registered_filter() {
+        final TestMessageBus bus = busBuilder().addFilter(new FailingFilter())
+                                               .build();
+
+        assertBusPostErrs(bus, ErrorType.FAILING_FILTER);
+    }
+
+    @Test
+    public void apply_registered_filters() {
+        final PassingFilter passingFilter = new PassingFilter();
+        final PassingFilter passingFilter2 = new PassingFilter();
+
+        final TestMessageBus bus = busBuilder().addFilter(passingFilter)
+                                               .addFilter(passingFilter2)
+                                               .addFilter(new FailingFilter())
+                                               .build();
+
+        assertBusPostErrs(bus, ErrorType.FAILING_FILTER);
+
+        assertTrue(passingFilter.passed());
+        assertTrue(passingFilter2.passed());
+    }
+
+    /**
+     * Asserts that bus acknowledges the error when posting a single message.
+     */
+    private static void assertBusPostErrs(TestMessageBus bus, ErrorType type) {
```

Comment:
`...Errors`?

Reply:
I can change it, but what I did mean to say [_errs_](https://www.merriam-webster.com/dictionary/err).